presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetTypeUtils.java

```diff
@@ -26,17 +26,15 @@ private ParquetTypeUtils()
     public static parquet.schema.Type getParquetType(HiveColumnHandle column, MessageType messageType, boolean useParquetColumnNames)
     {
         if (useParquetColumnNames) {
-            if (messageType.containsField(column.getName())) {
-                return messageType.getType(column.getName());
-            }
-            // parquet is case-sensitive, but hive is not. all hive columns get converted to lowercase
-            // check for direct match above but if no match found, try case-insensitive match
-            for (Type type : messageType.getFields()) {
-                if (type.getName().equalsIgnoreCase(column.getName())) {
-                    return type;
-                }
+            String name = column.getName();
+            Type type = getParquetTypeByName(name, messageType);
+
+            // when a parquet field is a hive keyword we append an _ to it in hive. When doing
```

Comment:
What happens if the field really contains a trailing _?

For minimal git diff and possible conflict, I suggest leaving the original Presto code here, and call our own custom method which checks these in order
1. Ignore case
2. Strip _ and then ignore case